src/item.h

```diff
@@ -1229,6 +1229,15 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
             /** flags change behavior of gun mode and are **not** equivalent to item flags */
             std::set<std::string> flags;
 
+            gun_mode( const std::string &mode = "", item *target = nullptr, int qty = 0 ) :
```

Comment:
Good point, but at least one constructor has to be here to mend [this](https://github.com/CleverRaven/Cataclysm-DDA/pull/17098/files#diff-f7ca61cf749eba039da961e8a7be1630L4260) line (I expect it has the same problem).